fix-point/app/Main.hs

```diff
@@ -1,15 +1,77 @@
 module Main where
 
+import Data.Semigroup ((<>))
+import Options.Applicative (help, Parser, ParserInfo, info, (<**>), header,
+                            helper, execParser, (<|>), metavar, long, short,
+                            fullDesc, progDesc, argument, str)
 import System.Directory (doesDirectoryExist)
-import System.Environment (getArgs)
 
 import CopyFiles (copyAll)
 import Rename
 
 main :: IO ()
-main = do
-    args <- getArgs
-    isDirectory <- doesDirectoryExist "three-layer"
-    if isDirectory
-    then copyAll "three-layer" args!!0 args!!1
-    else putStrLn "three-layer directory does not exist"
+main = run =<< execParser (args `withInfo` "Create project with given name")
+
+run :: Args -> IO ()
+run (Args input) = do
+    case input of
+        Project  proj           -> copyAll "three-layer" proj proj
+        ProjPref proj pref      -> copyAll "three-layer" proj pref
+        StringFP path           -> copyAll path "NEWNAME" "NEWDIR"
+        Combo    path proj pref -> copyAll path proj pref
+
+type Project    = String
+type Prefix     = String
+type SourcePath = FilePath
+
+data FPInput = Project Project
+             | ProjPref Project Prefix
+             | StringFP SourcePath
+             | Combo SourcePath Project Prefix
```

Comment:
I think it will be much easier if your data type will look like this:

```haskell
data Options = Options
    { projectName :: String
    , prefixName :: Maybe String
    , sourceDirectory :: Maybe String
    }
```

Imagine having 10 command-line options. It's impossible to have all all combinations of 10 options as separate constructors.

If you have `Parser a` then it's quite easy to get `Parser (Maybe a)` with `optional` combinator from alternative:

* https://hackage.haskell.org/package/base-4.11.1.0/docs/Control-Applicative.html#v:optional